File: src/main/java/com/wikibooks/spark/ch8/TokenizerSample.java
package com.wikibooks.spark.ch8;

import org.apache.spark.ml.feature.Tokenizer;
import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.RowFactory;
import org.apache.spark.sql.SparkSession;
import org.apache.spark.sql.types.DataTypes;
import org.apache.spark.sql.types.StructField;
import org.apache.spark.sql.types.StructType;

import java.util.Arrays;

public class TokenizerSample {

  public static void main(String[] args) throws Exception {

    SparkSession spark = SparkSession.builder()
            .appName("TokenizerSample")
            .master("local[*]")
            .getOrCreate();

    StructField sf1 = DataTypes.createStructField("input", DataTypes.StringType, true);
    StructType st1 = DataTypes.createStructType(Arrays.asList(sf1));
    Row r1 = RowFactory.create("Tokenization is the process");
    Row r2 = RowFactory.create("Refer to the Tokenizer");

    Dataset<Row> inputDF = spark.createDataFrame(Arrays.asList(r1, r2), st1);
    Tokenizer tokenizer = new Tokenizer().setInputCol("input").setOutputCol("output");
    Dataset<Row> outputDF = tokenizer.transform(inputDF);
    outputDF.printSchema();
    outputDF.show(false);

    spark.stop();
  }
}

File: src/main/scala/com/wikibooks/spark/ch8/scala/TokenizerSample.scala
package com.wikibooks.spark.ch8.scala

import org.apache.spark.ml.feature.Tokenizer
import org.apache.spark.sql.SparkSession

object TokenizerSample {

  def main(args: Array[String]) {

    val spark = SparkSession
      .builder()
      .appName("TokenizerSample")
      .master("local[*]")
      .getOrCreate()

    val data = Seq("Tokenization is the process", "Refer to the Tokenizer").map(Tuple1(_))
    val inputDF = spark.createDataFrame(data).toDF("input")
    val tokenizer = new Tokenizer().setInputCol("input").setOutputCol("output")
    val outputDF = tokenizer.transform(inputDF)
    outputDF.printSchema()
    outputDF.show(false)

    spark.stop
  }
}